cc/tf/rttops/math_ops.cc

```diff
@@ -208,53 +215,64 @@ REGISTER_OP("RttAddN")
     .SetIsAggregate();
 
 REGISTER_OP("RttMatmul")
+#if ROSETTA_ENABLES_SHAPE_INFERENCE
+  .SetShapeFn(::tensorflow::shape_inference::MatMulShape)
+#endif
   .Input("x: string")
   .Input("y: string")
   .Output("res: string")
   .Attr("transpose_a: bool = false")
   .Attr("transpose_b: bool = false");
-  // .SetShapeFn(::tensorflow::shape_inference::MatMulShape);
 
 REGISTER_OP("RttSquare").Input("x: string").Output("res: string");
 
-
 REGISTER_OP("RttReduceMean")
+#if ROSETTA_ENABLES_SHAPE_INFERENCE
```

Comment:
Please move this into the end of this MACRO.

Reply:
REGISTER_OP("RttReduceMean")
  .Input("input: string")
  .Input("reduction_indices: Tidx")
  .Output("output: string")
  .Attr("keep_dims: bool = false")
  .Attr("Tidx: {int32, int64} = DT_INT32")
#if ROSETTA_ENABLES_SHAPE_INFERENCE
  .SetShapeFn(::tensorflow::shape_inference::ReductionShape)
#endif
;
change it in this way?

Reply:
yes